pinecone/models.go

```diff
@@ -68,47 +68,47 @@ const (
 )
 
 type PodSpecMetadataConfig struct {
-	Indexed *[]string
+	Indexed *[]string `json:"indexed,omitempty"`
 }
 
 type PodSpec struct {
-	Environment      string
-	PodType          string
-	PodCount         int32
-	Replicas         int32
-	ShardCount       int32
-	SourceCollection *string
-	MetadataConfig   *PodSpecMetadataConfig
+	Environment      string                 `json:"environment"`
+	PodType          string                 `json:"pod_type"`
+	PodCount         int32                  `json:"pod_count"`
+	Replicas         int32                  `json:"replicas"`
+	ShardCount       int32                  `json:"shard_count"`
+	SourceCollection *string                `json:"source_collection,omitempty"`
+	MetadataConfig   *PodSpecMetadataConfig `json:"metadata_config,omitempty"`
 }
 
 type ServerlessSpec struct {
-	Cloud  Cloud
-	Region string
+	Cloud  Cloud  `json:"cloud"`
+	Region string `json:"region"`
 }
 
 type Vector struct {
-	Id           string
-	Values       []float32
-	SparseValues *SparseValues
-	Metadata     *Metadata
+	Id           string        `json:"id"`
+	Values       []float32     `json:"values,omitempty"`
```

Comment:
I thought we always had to have dense values, no? Like, you can technically "only" have sparse values in an index if you upload a dense vector that's just filled with `0`s and has a dimension of `1` or `2` or something, but I think you _have_ to have a dense value (so I think `omitempty` here is not good)

Reply:
(b/c you should never have an empty dense vector value)

Reply:
Makes sense, I also think empty slices default to `[]` in JSON when marshaling in Go, so that should be fine.

Reply:
You do have to have dense values, yes. Similar to the pointer issue I mentioned here: https://github.com/pinecone-io/go-pinecone/pull/21#discussion_r1591437117 I was mainly doing this defensively to avoid instances where we'd have `values: null` in the JSON, but I don't think that's ever expected to happen.

Reply:
Also the generated code that we map between uses `omitempty` for this, so I think it's fine to leave it as-is for now: https://github.com/pinecone-io/go-pinecone/blob/32391bcb7b862570139e272797ade4ecca50aab3/internal/gen/data/vector_service.pb.go#L88